hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotRenameRequest.java

```diff
@@ -0,0 +1,243 @@
+
+    final String volumeName = request.getVolumeName();
+    final String bucketName = request.getBucketName();
+    final String snapshotNewName = request.getSnapshotNewName();
+    final String snapshotOldName = request.getSnapshotOldName();
+
+    SnapshotInfo snapshotOldInfo = null;
+
+    try {
+      // Acquire bucket lock
+      mergeOmLockDetails(
+          omMetadataManager.getLock().acquireWriteLock(BUCKET_LOCK,
+                                                       volumeName, bucketName));
+      acquiredBucketLock = getOmLockDetails().isLockAcquired();
+
+      mergeOmLockDetails(omMetadataManager.getLock().acquireWriteLock(SNAPSHOT_LOCK,
+                                                       volumeName, bucketName, snapshotOldName));
+      acquiredSnapshotOldLock = getOmLockDetails().isLockAcquired();
+
+      mergeOmLockDetails(omMetadataManager.getLock().acquireWriteLock(SNAPSHOT_LOCK,
+                                                       volumeName, bucketName, snapshotNewName));
+      acquiredSnapshotNewLock = getOmLockDetails().isLockAcquired();
+
+      // Retrieve SnapshotInfo from the table
+      String snapshotNewTableKey = SnapshotInfo.getTableKey(volumeName, bucketName, snapshotNewName);
+
+      if (omMetadataManager.getSnapshotInfoTable().isExist(snapshotNewTableKey)) {
+        throw new OMException("Snapshot with name " + snapshotNewName + "already exist",
+            FILE_ALREADY_EXISTS);
+      }
+
+      // Retrieve SnapshotInfo from the table
+      String snapshotOldTableKey = SnapshotInfo.getTableKey(volumeName, bucketName,
+                                                 snapshotOldName);
+      snapshotOldInfo =
+          omMetadataManager.getSnapshotInfoTable().get(snapshotOldTableKey);
+
+      if (snapshotOldInfo == null) {
+        // Snapshot does not exist
+        throw new OMException("Snapshot with name " + snapshotOldName + "does not exist",
+            FILE_NOT_FOUND);
+      }
+
+      switch (snapshotOldInfo.getSnapshotStatus()) {
+      case SNAPSHOT_DELETED:
+        throw new OMException("Snapshot is already deleted. "
+            + "Pending reclamation.", FILE_NOT_FOUND);
+      case SNAPSHOT_ACTIVE:
+        break;
+      default:
+          // Unknown snapshot non-active state
+        throw new OMException("Snapshot exists but no longer in active state",
+            FILE_NOT_FOUND);
+      }
+
+      snapshotOldInfo.setName(snapshotNewName);
+
+      omMetadataManager.getSnapshotInfoTable().addCacheEntry(
+          new CacheKey<>(snapshotOldTableKey),
+          CacheValue.get(termIndex.getIndex()));
+
+      omMetadataManager.getSnapshotInfoTable().addCacheEntry(
+          new CacheKey<>(snapshotNewTableKey),
+          CacheValue.get(termIndex.getIndex(), snapshotOldInfo));
+
+      updateSnapshotInChainManager(omMetadataManager, snapshotOldInfo);
+
+      ozoneManager.getOmSnapshotManager().getSnapshotCache().invalidate(snapshotOldTableKey);
```

Comment:
1. I don't think it is good idea to invalidate the cache because of snapshot rename. I guess better approach would be to use snapshotId as key for snapshotCache.
2. Unit test is failing due to NPE in this line. Please take a look of that.
```
Error:  Tests run: 13, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.973 s <<< FAILURE! -- in org.apache.hadoop.ozone.om.request.snapshot.TestOMSnapshotRenameRequest
Error:  org.apache.hadoop.ozone.om.request.snapshot.TestOMSnapshotRenameRequest.testValidateAndUpdateCache -- Time elapsed: 0.178 s <<< ERROR!
java.lang.NullPointerException
	at org.apache.hadoop.ozone.om.request.snapshot.OMSnapshotRenameRequest.validateAndUpdateCache(OMSnapshotRenameRequest.java:192)
	at org.apache.hadoop.ozone.om.request.OMClientRequest.validateAndUpdateCache(OMClientRequest.java:145)
	at org.apache.hadoop.ozone.om.request.snapshot.TestOMSnapshotRenameRequest.testValidateAndUpdateCache(TestOMSnapshotRenameRequest.java:217)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at java.util.ArrayList.forEach(ArrayList.java:1259
```